app/js/model/list_model.js

```diff
@@ -4,66 +4,57 @@ import { Model } from 'components/fxos-mvc/dist/mvc';
-        icon: 'https://callahad.github.io/fxos-tabswitcher-addon/src/icon-128x128.png',
-        author: 'Dan Callahan',
-        type: 'addon',
-        url: 'https://github.com/callahad/fxos-tabswitcher-addon',
-        revision: '7096c0545980e250ee5be505b746421ffb9b19ce'
-      },
-      'http://mikehenrty.github.io/fxos-cracked/manifest.webapp': {
-        manifestURL: 'http://mikehenrty.github.io/fxos-cracked/manifest.webapp',
-        name: 'cracked',
-        description: 'A virtual cracked screen for your FirefoxOS device.',
-        icon: 'http://mikehenrty.github.io/fxos-cracked/style/icons/128.png',
-        author: 'David Flanagan',
-        type: 'addon',
-        url: 'https://github.com/mikehenrty/fxos-cracked',
-        revision: '16016c86f464c23e0465b22d43c08f52c1cc7349'
-      },
-      'https://callahad.github.io/disconnected-ensemble-src/src/manifest.webapp': {
-        manifestURL: 'https://callahad.github.io/disconnected-ensemble-src/src/manifest.webapp',
-        name: 'disconnected ensemble',
-        description: 'An app for Firefox OS that lets you share a number of musical toys that can run in your friends\' devices, served from your phone, either by typing in the IP address of the phone server, or by bumping NFC devices together.',
-        icon: 'https://callahad.github.io/disconnected-ensemble-src/src/img/icons/icon128x128.png',
-        author: 'soledad penadés',
-        type: 'packaged',
-        url: 'https://github.com/callahad/disconnected-ensemble-src',
-        revision: '94ad2d961aa5a902fe8150b280a38f48d84607eb'
+    return new Promise((resolve, reject) => {
+      Promise.all([this.loadLocalApps(), this.loadRemoteApps()]).then(
+      sources => {
+        var localApps = sources[0];
+        var remoteApps = sources[1];
+
+        // Try to fetch the remote app list, but if it fails, use the packaged
+        // one instead.
+        resolve(Object.keys(remoteApps).length ? remoteApps : localApps);
+      });
+    });
+  }
+
+  loadLocalApps() {
+    return new Promise((resolve, reject) => {
+      // Load from a previously networked session, if we have one.
+      var apps = localStorage.getItem('apps');
+      if (apps) {
+        resolve(JSON.parse(apps));
+        return;
       }
-    };
+
+      // Load from the package if all else fails.
+      var iframe = document.createElement('iframe');
+      iframe.src = 'apps.json';
+      iframe.hidden = true;
+      iframe.onload = () => {
+        var apps = JSON.parse(iframe.contentDocument.body.textContent);
+        document.body.removeChild(iframe);
+        resolve(apps);
+      };
+      document.body.appendChild(iframe);
+    });
+  }
+
+  loadRemoteApps() {
+    return new Promise((resolve, reject) => {
+      var xhr = new XMLHttpRequest({ mozSystem: true, mozAnon: true });
+      xhr.open('GET', 'http://directory.fxosapps.org/apps.json', true);
+      xhr.responseType = 'json';
+      xhr.onload = () => {
```

Comment:
We should also check the status code here, since the server could return a 500 or something, and in that case we should resolve with an empty object.